engine/movie/opening_cutscene.asm

```diff
@@ -0,0 +1,1235 @@
+
+	ld a, %11100100
+	ldh [rBGP], a
+	ld a, %11100000
+	ldh [rOBP0], a
+	call Intro_InitOmanyte
+	ret
+
+IntroScene2: ; 444a
+	call Intro_UpdateLYOverrides
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .skip_intro
+	dec [hl]
+	call Intro_InitBubble
+	ret
+
+.skip_intro
+	ld [hl], $10
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+
+IntroScene3:
+; rise towards the surface
+	call IntroScene3_Jumper
+	call IntroScene3_ScrollToSurface
+	ret nc
+; next scene if carry flag is set
+	call Intro_ResetLYOverrides
+	ld hl, hSCY
+	inc [hl]
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+
+IntroScene4:
+; at surface; Lapras surfs to left of screen
+	ld a, [wIntroSpriteStateFlag]
+	and a
+	jr nz, .next
+	ld hl, wIntroFrameCounter2
+	inc [hl]
+	ld a, [hl]
+	and $07
+	jr nz, .skip_move_left
+	ld hl, hSCX
+	dec [hl]
+
+.skip_move_left
+	call Intro_AnimateOceanWaves
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	xor a
+	ld [wIntroFrameCounter1], a
+
+IntroScene5:
+; fade out
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	inc [hl]
+	swap a
+	and $f
+	ld e, a
+	ld d, 0
+	ld hl, .palettes
+	add hl, de
+	ld a, [hl]
+	cp -1
+	jr z, .next
+	ldh [rBGP], a
+	call Intro_AnimateOceanWaves
+	ld hl, hSCX
+	dec [hl]
+	dec [hl]
+	ret
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ret
+
+.palettes:
+	db %11100100
+	db %11100100
+	db %10010000
+	db %01000000
+	db %00000000
+	db -1
+
+IntroScene17:
+; delay a bit before leading into the title screen
+	ld c, 64
+.loop
+	call DelayFrame
+	dec c
+	jr nz, .loop
+; set done flag
+	ld hl, wIntroJumptableIndex
+	set 7, [hl]
+	ret
+
+IntroScene3_ScrollToSurface:
+	ld hl, wIntroFrameCounter2
+	inc [hl]
+	ld a, [hl]
+	and %00000011
+	jr nz, .skip_move_left
+	ld hl, hSCX
+	dec [hl]
+
+.skip_move_left
+	and 1
+	jr nz, .no_carry
+	ld hl, wGlobalAnimYOffset
+	inc [hl]
+	ld hl, hSCY
+	ld a, [hl]
+	dec [hl]
+	and $f
+	call z, Intro_UpdateTilemapAndBGMap
+	ld a, [wIntroFrameCounter1]
+	and a
+	jr z, .carry
+
+.no_carry
+	and a
+	ret
+
+.carry
+	scf
+	ret
+
+IntroScene3_Jumper:
+	jumptable .subroutines, wIntroFrameCounter1
+.subroutines
+	dw .scene3_2
+	dw .scene3_2
+	dw .scene3_2
+	dw .scene3_1
+	dw .scene3_2
+	dw .scene3_2
+	dw .scene3_3
+	dw .scene3_3
+	dw .scene3_3
+	dw .scene3_4
+	dw .scene3_5
+	dw .scene3_6
+	dw .scene3_6
+	dw .scene3_6
+	dw .scene3_6
+	dw .scene3_6
+	dw .scene3_6
+
+.scene3_1:
+	call Intro_InitLapras
+	ld a, %11100100
+	ldh [rOBP0], a
+
+.scene3_2:	; fallthrough
+	call Intro_AnimateOceanWaves
+	ret
+
+.scene3_3:
+	call Intro_InitMagikarps
+	call Intro_AnimateOceanWaves
+	ret
+
+.scene3_4:
+	ld hl, wIntroFrameCounter2
+	ld a, [hl]
+	and %00011111
+	jr z, .load_palettes
+	call Intro_InitMagikarps
+	ret
+.load_palettes
+	callab Function962d	; load magikarp palettes
+	ret
+
+.scene3_5:
+	xor a
+	ldh [hLCDCPointer], a
+	ret
+
+.scene3_6:
+	call Intro_UpdateLYOverrides
+	ret
+
+Intro_InitBubble:
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and $0f
+	ret nz
+
+	ld a, [hl]
+	and $70
+	swap a
+	ld e, a
+	ld d, $00
```

Comment:
Just `ld d, 0`; there's often "`ld e, a / ld d, 0`" or "`ld c, a / ld b, 0`" to do "`de := a`"` or "`bc := a`"